### DISCUSSION TITLE
How to toggle spatial mesh observer visibility?

### DISCUSSION THREAD

**Discussion by justinpando** [microsoft/MixedRealityToolkit-Unity#9673] [Q&A] [answered]
I'm following along with [Create dynamic content using Solvers](https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unity/tutorials/mr-learning-base-05) and reached this Tip - 
> If you want to see the Spatial Awareness Mesh when testing the Tap To Place functionality on your HoloLens, you can temporarily set the Spatial Mesh Observer's Display Option to Visible. For a reminder on how to change the Display Option, you can refer to the Changing the Spatial Awareness Display Option instructions.

The issue is that the guide only shows you how to set the mesh visibility by creating a new custom profile and toggling the setting off in that profile. I'm not seeing a way to procedurally change mesh visibility at runtime - how is this meant to be done? In the tutorial would have logically expected to be able to add functions to OnPlacingStarted and OnPlacingStopped.

**Comment by MaxWang-MS** [comment, marked as answer]
Please take a look here: https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/spatial-awareness/usage-guide#showing-and-hiding-the-spatial-mesh
Note other docs under the Spatial Awareness folder may be also helpful to you.

**Reply by justinpando** [reply]
Thank you!
